src/hooks/createUseApi.js

```diff
@@ -0,0 +1,333 @@
+// @flow
+import invariant from 'invariant'
+import { capitalize, uniqueId } from 'lodash'
+import { useCallback, useEffect, useRef, useState, useMemo } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import shallowEqual from 'react-redux/lib/utils/shallowEqual'
+
+import createActionCreators from '../actions'
+import {
+  type ApiTypeMap,
+  type MethodName,
+  type Query,
+  type RequestAction,
+  type RequestStatus,
+} from '../flowTypes'
+import { getIdAttribute } from '../types'
+import { getEntityState, getRequestState, stringifyQuery } from '../utils'
+
+type BaseOptions = {|
+  method?: MethodName,
+  requestParams?: Query,
+  refresh?: any,
+  denormalize?: boolean,
+  lazy?: boolean,
+  onSuccess?: () => void,
+  onFailure?: () => void,
+|}
+
+type QueryOptions = {|
+  ...BaseOptions,
+  query: Query,
+|}
+
+type IdOptions = {|
+  ...BaseOptions,
+  id: string,
+|}
+
+type ResolvedOptions = {|
+  method: MethodName,
+  query: Query,
+  requestParams: Query,
+  lazy: boolean,
+  denormalize: boolean,
+
+  refresh?: any,
+
+  onSuccess?: () => void,
+  onFailure?: () => void,
+|}
+
+type Options = BaseOptions | QueryOptions | IdOptions
+
+function createUseApi(apiTypes: ApiTypeMap) {
+  const actionCreators = createActionCreators(apiTypes)
+
+  return function useApi<Body, Value>(
+    entityType: $Keys<ApiTypeMap>,
+    options?: Options
+  ): [RequestStatus<Value>, RequestAction<Body>] {
+    const {
+      query,
+      method,
+      requestParams,
+      refresh,
+      denormalize,
+      lazy,
+
+      onSuccess,
+      onFailure,
+    } = resolveOptions(apiTypes, entityType, options)
+
+    invariant(
+      apiTypes[entityType],
+      `Invalid type value '${entityType}' passed to \`connect\` (expected one of: ${Object.keys(
+        apiTypes
+      ).join(', ')})`
+    )
+
+    const resolvedType = apiTypes[entityType]
+
+    invariant(
+      typeof resolvedType[method] === 'function',
+      `Invalid method "${method}" specified for api type "${entityType}"`
+    )
+
+    const [elementId] = useState(uniqueId())
```

Comment:
```suggestion
    const [elementId] = useState(() => uniqueId())
```

Just so that it doesn't actually do that all the time.